AirCasting/Utils/Publishers/LogErrorAndCompletePublisher.swift

```diff
@@ -0,0 +1,41 @@
+// Created by Lunar on 16/06/2021.
+//
+
+import Combine
+
+extension Publisher {
+    /// Logs any error ending the upstream using provided logging function *and* tears downstream as completed.
+    /// - Parameters:
+    ///   - message: A message that will prefix the `error.localizedDescription` printout.
+    ///   - logFunc: Optional custom logging function.
+    /// - Returns: A publisher that matches the upstream `Output` and has a `Failure` set to `Never`.
+    func logErrorAndComplete(message: String = "", logFunc: @escaping (String) -> Void = { Log.error($0) }) -> Publishers.LogErrorAndComplete<Self> {
+        .init(upstream: self, logMessage: message, logFunc: logFunc)
+    }
+}
+
+extension Publishers {
+    struct LogErrorAndComplete<Upstream: Publisher>: Publisher {
```

Comment:
IMo if there is a need for sturct like that that means that something is wrong. Logs should be more contextual for what has tiggered it. log itself uses #file to get the current file to give more context to the logs and this will be useless in this file

Reply:
`#file` and `#line` can be passed here without a problem, didn't think of that TBH, will get fixed 👌. Those operators are by all means not **needed**, but I think are super handy and encourage logging in parts of the code that use `Combine`. In order to log errors now instead of this:
```Swift
publisher.
    .filter(someFilter)
    .map(someMap)
    .handleEvents(recieveCompletion: { result in
        if case .failure(let error) = result else { return }
        logger.error("Error in the pipeline! \(error.localizedDescription)")
    })
```

which is not-so-pretty,
you can do:
```Swift
publisher.
    .filter(someFilter)
    .map(someMap)
    .logError("Error in the pipeline!")
```